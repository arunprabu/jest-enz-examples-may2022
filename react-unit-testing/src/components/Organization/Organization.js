import React, { useState } from 'react';

const Organization = (props) => {
  const [ company, setCompany ] = useState('');

  const handleChange = (event) => {
    console.log(event.target.value);
    setCompany(event.target.value);
  }

  return (
    <div>
      <h2>Organization Info!</h2>
      <h3>Welcome to My Company's Official Website</h3>
      <p data-testid="orgEl">{props.orgName}</p>

      <input type="text" value={company} onChange={handleChange} data-testid="companyNameInput" />
    </div>
  )
}

export default Organization;