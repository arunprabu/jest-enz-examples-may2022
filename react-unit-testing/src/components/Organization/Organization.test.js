import { fireEvent, render, screen } from '@testing-library/react';
import ReactDOM from 'react-dom/client';
import Organization from "./Organization";
import renderer from 'react-test-renderer';

describe('Organization', () => { 

  // test case #1 -- renders the comp properly
  // it('renders the organization comp properly', () => {
  //   const root = ReactDOM.createRoot(document.createElement("div"));
  //   root.render(<Organization />);
  // });

  // test case #2 -- testing the prop 
  it('should display the props correctly', () => {
    // take the element to put under test
    render(<Organization orgName="Cognizant Pvt Ltd"/>);
    // fetching the element by data-testid orgEl //refer Organization.js file  
    // checking if the props are received correctly 
    expect(screen.getByTestId('orgEl')).toHaveTextContent('Cognizant Pvt Ltd');
  });

  // test case #3 
  it('should have Welcome to My Company\'s Official Website as text', () => {
    render(<Organization />);
    const officialTxtEl = screen.getByText('Welcome to My Company\'s Official Website');
    expect(officialTxtEl).toBeInTheDocument();
  });

  // test case #4 -- to test events in react
  it('should update text input when onChange occurs', () => {
    render(<Organization />);
    // finding the element using getByTestId 'companyNameInput'
    const companyNameInputEl =  screen.getByTestId('companyNameInput');
    // now check whether onChange working or not 
    // mock fire change event -- with the value 
    fireEvent.change(companyNameInputEl, { target: {value: 'test123'}});
    // finally check if the input is getting the updated data
    expect(companyNameInputEl.value).toBe('test123');
  });

  // snapshot testing
  it('should have right company comp snapshot', () => {
    // to take snapshot we need a tool 'react-test-renderer' -- npm i react-test-renderer -D 
    const snapShotTree = renderer.create(<Organization orgName="Cognizant Pvt Ltd"/> ).toJSON();
   // console.log(snapShotTree);
    expect(snapShotTree).toMatchSnapshot();
  });

});

