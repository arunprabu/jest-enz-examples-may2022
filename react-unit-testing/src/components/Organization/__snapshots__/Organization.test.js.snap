// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organization should have right company comp snapshot 1`] = `
<div>
  <h2>
    Organization Info!
  </h2>
  <h3>
    Welcome to My Company's Official Website
  </h3>
  <p
    data-testid="orgEl"
  >
    Cognizant Pvt Ltd
  </p>
  <input
    data-testid="companyNameInput"
    onChange={[Function]}
    type="text"
    value=""
  />
</div>
`;
